src/server/conn/mod.rs

```diff
@@ -5,8 +5,14 @@
 //! are not handled at this level. This module provides the building blocks to
 //! customize those things externally.
 //!
-//! This module is split by HTTP version. Both work similarly, but do have
-//! specific options on each builder.
+//! This module is split by HTTP version, providing a connection builder for
+//! each. They work similarly, but they each have specific options.
+//!
+//! If your server needs to support both versions, an auto-connection builder is
+//! provided in the [`hyper-util`](https://github.com/hyperium/hyper-util/tree/master)
+//! crate. This builder wraps the HTTP/1 and HTTP/2 connection builders from this
+//! module, allowing you to set configuration for both. The builder will then check
+//! the version of the incoming connection and serve it accordingly.
```

Comment:
This should link to the hyper-util docs when those are generated. Perhaps we should open an issue for creating an example for auto-connection in hyper-util as well?